core/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -415,6 +416,14 @@ public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) t
             builder.field(Fields._NODE, shard.getNodeIdText());
         }
         if (nestedIdentity != null) {
+            if (nestedIdentity.isIncludeIdAndIndex()) {
```

Comment:
Can you add a randomization to SearchHitTests#createTestItem() that randomly sets includeIdAndIndex on the nestedIdentity, so we are sure we test parsing this?